File: 003.pl
#!usr/bin/perl

use strict;
use bigint;

#my @primes = &prime_factors(600851475143);
my @primes = &prime_factors(12345678987654321);
print "@primes\n";

sub prime_factors{
	my $number = @_[0];	
	my @factors;
	my $divisor = 2;
	while ($number > 1){
		while (($number % $divisor) == 0){
			push (@factors, $divisor);
			$number = ($number/$divisor);
		}
		$divisor += 1;
	}
	return @factors;
}

File: 003.c
#include <stdio.h>

int
main() {
  long n = 600851475143;
  int d = 2;

  while (n > 1) {
    while (n % d == 0) {
      printf("Divisor: %d\n", d);
      n /= d;
    }
    d++;
  }
  return 0;
}
